### ISSUE TITLE
Existing cluster contexts/logins are removed when a new cluster is deployed

### ISSUE THREAD

**Issue by marcusbooyah** [nforgeio/neonKUBE#1888]
All my logins in kubeconfig got deleted when installing a new cluster

**Comment by jefflill** [comment]
I've been able to replicated this.

**Comment by jefflill** [comment]
Actually, I **cannot replicate this**.

In my first attempt, I created a tiny hyper-v cluster using my cheats and then I tweaked the cluster definition and my cheat script and created a second cluster, but I forgot to edit the line where I delete any existing cluster, so that removed the first cluster I created and then my script created the new cluster.  The old cluster (and login) were deleted because I (unintentionally) deleted it.

I tried this again by creating two clusters (and taking care not to delete the first one) and both clusters and logins are present.

I've also had the headend cluster in my config for a long time and have been able to create/delete clusters freely without removing the headend cluster.

@marcusbooyah: Is it possible that this was caused by: https://github.com/nforgeio/neonCLOUD/issues/437

If not, could you give me some more details:

* What was the target environment for the new cluster
* What kind of contexts were removed?  NeonKUBE clusters and/or the Headend context (not that the Headend context is not treated as a NeonKUBE context because it's format is very old).
* By change, did you do what I did and accidentally deleted the first cluster and context?

**Comment by jefflill** [comment]
@marcusbooyah told me that the cluster login removed was for our headend cluster, which is not currently considered to be a NeonKUBE cluster (because it's pretty old).

I tried deploying a new cluster with the headend cluster in the config and I ended up with both clusters in the kubeconfig file.

I tried again, manually causing the cluster deployment to fail to see if that the recovery from that would cause the headend cluster liogin to be deleted. but it was still there afterwards.

I did find and fix a problem related to saving the kubeconfig, but I don't believe that fixed this problem.

I'm going to keep this issue open so we can watch for this, but I'm at a dead-end right now.
